model.py

```diff
@@ -800,12 +893,13 @@ def _run_one_epoch(data_loader, callbacks, mode):
                 for metric in self._metrics:
                     res = metric.accumulate()
                     metrics.extend(to_list(res))
+                    
                 assert len(metrics_name) == len(metrics)
                 for k, v in zip(metrics_name, metrics):
                     logs[k] = v
 
                 logs['step'] = step
-                logs['batch_size'] = data[0].shape[0]
+                logs['batch_size'] = batch_size
```

Comment:
It seems  metrics result are gathered result, should `batch_size` also be summed up for multi-devices

Reply:
done, thanks.